### ISSUE TITLE
`Style/MapToHash` autocorrect breaks multiline code

### ISSUE THREAD

**Issue by ojab** [rubocop/rubocop#10313]
Consider the code:
```ruby
{foo: bar}
  .map { |k, v| [k.to_s, v.do_something] }
  .to_h
  .freeze
```
and `.rubocop.yml`: 
```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.0.3
```
then
```
rubocop /tmp/1.rb -A
Inspecting 1 file
E

Offenses:

/tmp/1.rb:2:4: C: [Corrected] Style/MapToHash: Pass a block to to_h instead of calling map.to_h.
  .map { |k, v| [k.to_s, v.do_something] }
   ^^^
/tmp/1.rb:3:1: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
/tmp/1.rb:4:3: E: Lint/Syntax: unexpected token tDOT
(Using Ruby 3.0 parser; configure using TargetRubyVersion parameter, under AllCops)
  .freeze
  ^

1 file inspected, 3 offenses detected, 2 offenses corrected
```
with new code
```ruby
{foo: bar}
  .to_h { |k, v| [k.to_s, v.do_something] }
  
  .freeze
```

## RuboCop version
```
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 3.0.3 x86_64-linux)
  - rubocop-performance 1.12.0
  - rubocop-rails 2.12.4
  - rubocop-rspec 2.6.0
 ```
